apps/factory/src/components/prompt_output.tsx

```diff
@@ -3,24 +3,26 @@ import Box from "@mui/material/Box";
 import IconButton from "@mui/material/IconButton";
 import Typography from "@mui/material/Typography";
 import SvgIcon from "@mui/material/SvgIcon";
-import { ModelTypeSchema } from "~/generated/prisma-client-zod.ts";
+import {
+  ModelTypeSchema,
+  ModelTypeType,
+} from "~/generated/prisma-client-zod.ts";
+import ModelTypeRenderer from "./model_type_renderer";
 
 const PromptOutput = ({
   output,
   modelType,
 }: {
   output: string;
-  modelType: string;
+  modelType: ModelTypeType;
 }) => {
   return (
     <Box>
-      {modelType === ModelTypeSchema.Enum.TEXT2TEXT ? (
-        <Typography variant="body2" textAlign={"left"}>
-          {output}
-        </Typography>
-      ) : (
-        <img className="h-48 w-48 object-fill " src={output} alt="Image" />
-      )}
+      <ModelTypeRenderer
```

Comment:
please rename this component to PromptCompletion, since most of the naming convention is related to that.

Reply:
sure